beacon-chain/state/state-native/v1/state_trie.go

```diff
@@ -210,7 +247,7 @@ func (b *BeaconState) initializeMerkleLayers(ctx context.Context) error {
 	if len(b.merkleLayers) > 0 {
 		return nil
 	}
-	fieldRoots, err := computeFieldRoots(ctx, b.state)
+	fieldRoots, err := computeFieldRoots(ctx, b.ToProtoUnsafe().(*ethpb.BeaconState))
```

Comment:
To prevent any unexpected panics, do an `ok` check when casting it to a proto pointer and return an error on it.